### PR TITLE
Feat/allow renewal of privileges

### LINKED ISSUE #295: Backend handles purchase of expired privileges
Currently, the system does not allow a licensee to purchase a privilege for the same jurisdiction twice, even if that previously purchased privilege is expired. The backend needs to handle the following case scenarios:

- Licensee has no active license (return 400 invalid request)
- Licensee has active license and is purchasing privilege for an existing privilege (preserve previous expired privilege, add new privilege)

As part of this ticket, we need to design how we will store previous privilege records so that it is returned along with the latest privilege record.

### PR DESCRIPTION
**Pull request by landonshumway-ia**
We need to allow licensees to renew privileges if their license expiration date has been updated. We still prevent them
from purchasing the same privilege if their license has not been updated since their last purchase.

As part of this change, we have updated the data model for privilege records to include a `dateOfRenewal` field, similar
to licenses. The sorting key for privilege and license records will now be suffixed with this 'dateOfRenewal' date stamp. So for example, supposing a privilege was purchased on 2024-11-08, this SK:
```
aslp#PROVIDER#privilege/ne
```

Will now be this tracked as this SK:
```
aslp#PROVIDER#privilege/ne#2024-11-08
```

### Requirements List
- No new requirements. The change should be backwards compatible with the existing schema.

### Description List
- Add logic to allow user to renew their existing privileges if their license expiration date has been updated.
- move common test resources into common python folder
- Update privilege and license sorting keys to include 'dateOfRenewal' date stamp suffix.

### Testing List
- `yarn test:unit:all` should run without errors or warnings
- `yarn serve` should run without errors or warnings
- `yarn build` should run without errors or warnings
- For API configuration changes: CDK tests added/updated in `backend/compact-connect/tests/unit/test_api.py`
- Code review

Closes #295

### CODE REVIEW

**Review by jusdino — APPROVED** [review]
This looks great!

**Comment by landonshumway-ia** [comment]
@jlkravitz This is now ready for CSG review.

**Comment by jlkravitz** [comment]
- [ ] review the pull request to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] skim all new code, in the context of existing code, looking for problems (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] methodically review all new tests for correctness, quality of naming
	- [ ] look at code coverage of tests
	- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] run a security audit of dependencies (e.g. `npm audit` and `pip audit`) to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)

File: backend/compact-connect/lambdas/purchases/handlers/privileges.py [hunk 1]
```diff
@@ -151,28 +153,32 @@ def post_purchase_privileges(event: dict, context: LambdaContext):  # noqa: ARG0
...
 
-    for privilege_jurisdiction in existing_privilege_jurisdictions:
-        if privilege_jurisdiction.lower() in selected_jurisdictions_postal_abbreviations:
+    existing_privileges = [record for record in user_provider_data['items'] if record['type'] == 'privilege']
+    # a licensee can only purchase an existing privilege for a jurisdiction
+    # if their existing privilege expiration date does not match their license expiration date
+    for privilege in existing_privileges:
+        if (
+            privilege['jurisdiction'].lower() in selected_jurisdictions_postal_abbreviations
+            and privilege['dateOfExpiration'] == license_record['dateOfExpiration']
```
Comment (jlkravitz) [inline backend/compact-connect/lambdas/purchases/handlers/privileges.py:178]: Not about the code, but mind giving me the quick tldr on this logic? A bit confused by the constraint.

**Review by jlkravitz — APPROVED** [review]
Sorry for the delay on this one- one small question but otherwise looks good!

File: backend/compact-connect/lambdas/purchases/handlers/privileges.py (hunk 1 above)
Comment (jlkravitz) [reply backend/compact-connect/lambdas/purchases/handlers/privileges.py:182]: Great, thanks!

**Review by isabeleliassen — APPROVED** [review]

**Review by jlkravitz — APPROVED** [review]

**Pull request merged**